connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -65,11 +65,25 @@ message SQL {
 message Read {
   oneof read_type {
     NamedTable named_table = 1;
+    FileDataSource file_data_source = 2;
   }
 
   message NamedTable {
     repeated string parts = 1;
   }
+
+  message FileDataSource {
+    repeated string path = 1;
+    string format = 2;
+    string schema = 3;
+    repeated DataSourceOptionPair options = 4;
```

Comment:
maybe simply `map<string, string>`?

Reply:
oh yes.. I forget that proto supports `MAP`. Will update this.